pkg/datasource.go

```diff
@@ -153,46 +220,94 @@ func (o *OCIDatasource) QueryData(ctx context.Context, req *backend.QueryDataReq
 	}
 }
 
+func (o *OCIDatasource) getConfigProvider(environment string, tenancymode string, req *backend.QueryDataRequest) error {
+	switch environment {
```

Comment:
Question: Can  you change `environment`/`tenancymode` on a per-query basis? If not would strongly suggest calling `getConfigProvider` in `NewOCIDatasource`. Or basically, can you populate `o.tenancyAccess` when datasource has been configured without having to take special measures depending on what query it is? Doing this will simplify a lot of things and you would be guaranteed by the grafana-plugin-sdk-go that the `NewOCIDatasource` is ever executed by one goroutine at a time and by that removing the need for synchronization/locks as I referred to in my earlier comment.